### DISCUSSION TITLE
nestjs-caching-redis

### DISCUSSION THREAD

**Discussion by giscus** [tomwray13/tomray.dev#51] [Blog Comments]
# nestjs-caching-redis

A detailed tutorial with NestJS examples on how to implement in-memory and Redis caching using the NestJS Cache Manager and Cache Interceptor

https://www.tomray.dev/nestjs-caching-redis

**Comment by Notalifeform** [comment]
nice write-up, thx. The tests seem currently broken.. any idea how to fix them?

```
 FAIL  src/pokemon/pokemon.service.spec.ts
  ● PokemonService › should be defined

    Nest can't resolve dependencies of the PokemonService (?, CACHE_MANAGER). Please make sure that the argument HttpService at index [0] is available in the RootTestModule context.

    Potential solutions:
    - If HttpService is a provider, is it part of the current RootTestModule?
    - If HttpService is exported from a separate @Module, is that module imported within RootTestModule?
      @Module({
        imports: [ /* the Module containing HttpService */ ]
      })


```

**Reply by Notalifeform** [reply]
I think this should be it: https://github.com/tomwray13/nestjs-caching/pull/1

**Reply by khatri-kelashk** [reply]
Note : As written in shared repository, Kindly import the "HttpModule" in PokemonModule, as PokemonService is consuming "HttpService" (from "@nestjs/axios" ).
PokemonModule file Code will be look like this:

import { HttpModule } from '@nestjs/axios';
import { Module } from '@nestjs/common';
import { PokemonController } from './pokemon.controller';
import { PokemonService } from './pokemon.service';

@module({
imports: [HttpModule],
controllers: [PokemonController],
providers: [PokemonService]
})
export class PokemonModule {}

**Comment by khatri-kelashk** [comment]
Note : As written in  shared repository, Kindly import the "HttpModule" in PokemonModule, as PokemonService is consuming "HttpService" (from "@nestjs/axios" ).
PokemonModule file Code will be look like this:

import { HttpModule } from '@nestjs/axios';
import { Module } from '@nestjs/common';
import { PokemonController } from './pokemon.controller';
import { PokemonService } from './pokemon.service';

@Module({
  imports: [HttpModule],
  controllers: [PokemonController],
  providers: [PokemonService]
})
export class PokemonModule {}

**Comment by denkyipatrick** [comment]
Really nice work. Thanks!
